docs/user/ppl/cmd/parse.rst

```diff
@@ -0,0 +1,97 @@
+=============
+parse
+=============
+
+.. rubric:: Table of contents
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+Description
+============
+| The ``parse`` command parses a text field using a regular expression and append the result to the search result.
+
+
+Syntax
+============
+parse <field> <regular-expression>
+
+* field: mandatory. The field must be a text field.
+* regular-expression: mandatory. The regular expression used to extract new fields from given text field. If a new field name already exists, it will replace the original field.
```

Comment:
I think regex is generic but not easy to use. Could we consider [logfmt](https://brandur.org/logfmt) or [grok](https://github.com/thekrakken/java-grok) at first?

Reply:
Yes we can add more methods to parse, probably would be `parse [method={regex|grok|...}] <field> <pattern>` and defaults to regex. Will do it in future PRs